clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -460,12 +442,14 @@ public Node provide() {
         private final String callName;
         private final long deadlineMs;
         private final NodeProvider nodeProvider;
+        private final int maxRetries;
         private int tries = 0;
 
-        Call(String callName, long deadlineMs, NodeProvider nodeProvider) {
+        Call(String callName, long deadlineMs, NodeProvider nodeProvider, int maxRetries) {
```

Comment:
This could just use `KafkaAdminClient#maxRetries`, right?  Unless we want to make this configurable per call, which I don't think this patch does.